### PR TITLE
 Implement tests for Services

### PR DESCRIPTION
**Pull request by SRozhina**

### CODE REVIEW

File: ITEventsTests/DateFormatterServiceTests.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    var dateFormatterService: DateFormatterService!
+    var dateIntervalSameDate: DateInterval!
+    var dateIntervalDifferentDates: DateInterval!
+    
+    override func setUp() {
+        super.setUp()
+        
+        dateFormatterService = DateFormatterService()
+
+        let dateFormatter: DateFormatter = {
```
Comment (AVykhrystyuk) [inline ITEventsTests/DateFormatterServiceTests.swift]:
```swift
let dateFormatter: DateFormatter = {
   let formatter = DateFormatter()
   formatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
   return formatter
}()
```
Looks like code above can be simplified to:
```swift
let dateFormatter = DateFormatter()
dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
```

File: ITEventsTests/DateFormatterServiceTests.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }()
+        
+        let startDate = "2018-01-18 19:00:00"
+        let endDate = "2018-01-18 22:00:00"
+        let endDifferentDate = "2018-01-19 22:00:00"
+        
+        dateIntervalSameDate = DateInterval(start: dateFormatter.date(from: startDate)! ,
+                                            end: dateFormatter.date(from: endDate)!)
+        dateIntervalDifferentDates = DateInterval(start: dateFormatter.date(from: startDate)!,
+                                                  end: dateFormatter.date(from: endDifferentDate)!)
```
Comment (AVykhrystyuk) [inline ITEventsTests/DateFormatterServiceTests.swift]: I'd do the following:
- removed extra variables (```startDate```, ```endDate```, ```endDifferentDate```)  as they only making a noise here 
- renamed the DateInterval variables to be more explicit

```swift     
sameDayDateInterval = DateInterval(start: dateFormatter.date(from:  "2018-01-18 19:00:00")! ,
                                   end: dateFormatter.date(from: "2018-01-18 22:00:00")!)
differentDaysDateInterval = DateInterval(start: dateFormatter.date(from: "2018-01-18 19:00:00")!,
                                         end: dateFormatter.date(from: "2018-01-19 22:00:00")!)

```

File: ITEventsTests/Info.plist [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AVykhrystyuk) [inline ITEventsTests/Info.plist]: I am not sure if this file is really needed? Can you please check?

File: ITEventsTests/SearchParametersServiceTests.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import XCTest
+@testable import ITEvents
+
+class SearchParametersServiceTests: XCTestCase {
+    var searchParametersService: SearchParametersService!
+    
+    override func setUp() {
+        super.setUp()
+        searchParametersService = SearchParametersService()
+
```
Comment (AVykhrystyuk) [inline ITEventsTests/SearchParametersServiceTests.swift]: extra empty line to be removed :)

File: ITEventsTests/DateFormatterServiceTests.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import XCTest
+@testable import ITEvents
+
+class DateFormatterServiceTests: XCTestCase {
+    var dateFormatterService: DateFormatterService!
```
Comment (AVykhrystyuk) [inline ITEventsTests/DateFormatterServiceTests.swift]: All your code in tests should work also with a protocol not with implementation:
```swfit
var dateFormatterService: IDateFormatterService!
```

File: ITEventsTests/SearchParametersServiceTests.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import XCTest
+@testable import ITEvents
+
+class SearchParametersServiceTests: XCTestCase {
+    var searchParametersService: SearchParametersService!
```
Comment (AVykhrystyuk) [inline ITEventsTests/SearchParametersServiceTests.swift]: All your code in tests should also work with a protocol not with implementation:
```swfit
var searchParametersService: ISearchParametersService!
```

And you will see why fixing this test

File: ITEventsTests/SelectedEventServiceTests.swift [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import XCTest
+@testable import ITEvents
+
+class SelectedEventServiceTests: XCTestCase {
+    var selectedEventService: SelectedEventService!
```
Comment (AVykhrystyuk) [inline ITEventsTests/SelectedEventServiceTests.swift]: All your code in tests should also work with a protocol not with implementation:
```swfit
var selectedEventService: ISelectedEventService!
```

File: Podfile.lock [hunk 8]
```diff
@@ -2,17 +2,18 @@ PODS:
...
-  - Reusable (4.0.2):
-    - Reusable/Storyboard (= 4.0.2)
-    - Reusable/View (= 4.0.2)
-  - Reusable/Storyboard (4.0.2)
-  - Reusable/View (4.0.2)
-  - Swinject (2.4.1)
-  - SwinjectStoryboard (2.0.1):
-    - Swinject (~> 2.4)
-  - TagListView (1.3.1)
+  - Reusable (4.0.0):
```
Comment (AVykhrystyuk) [inline Podfile.lock]: This file should be entirely reverted (just take file content from the master and copy/paste it here)

File: ITEventsTests/Info.plist (hunk 3 above)
Comment (SRozhina) [reply ITEventsTests/Info.plist]: It is default line. I can make changes only in dict

**Pull request merged**